### PR TITLE
Handle envoy RDS protocol error response - 1.1

### PR DESCRIPTION
**Pull request by knrc**
Istio-10105 for 1.1 branch

### CODE REVIEW

**Comment by rshriram** [comment]
@hzxuzhonghu is this same as your other PR?

**Comment by knrc** [comment]
@rshriram It doesn't look like it, I don't think that PR addresses the race condition so will still suffer from sidecars not being provided with their routes.

**Comment by hzxuzhonghu** [comment]
@rshriram Not same. 

@knrc Seems not only RDS, but also LDS/CDS/EDS have the same behavior.  And currently pilot does not react to ACK/NACK, it just record related metrics

**Comment by knrc** [comment]
@hzxuzhonghu +1, although the RDS failure is the one we can currently reproduce with our tests so it is easier for me to verify it is addressed.  The test case associated with this PR emulates the conditions for the race so we can likely do something similar with the others.

**Comment by knrc** [comment]
I have a slightly modified version of this to upload, we discovered another race while testing 1.0.x and I've just tracked it down to the same area.  I have a fix and will rebase/push tomorrow.

**Comment by knrc** [comment]
This PR has been updated and is good to go

**Comment by GregHanson** [comment]
@costinm there are separate PR to 1.0 and master branches:
https://github.com/istio/istio/pull/10107
https://github.com/istio/istio/pull/10109

**Comment by knrc** [comment]
+1 I submitted one for each as they were slightly different at the time.  The master one needs rebasing but I can handle that today.

File: pilot/pkg/proxy/envoy/v2/ads_test.go [hunk 1]
```diff
@@ -244,3 +251,184 @@ func TestAdsUpdate(t *testing.T) {
 	strResponse, _ = model.ToJSONWithIndent(res1, " ")
 	_ = ioutil.WriteFile(env.IstioOut+"/edsv2_update.json", []byte(strResponse), 0644)
 }
+
+func TestEnvoyRDSProtocolError(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (knrc) [reply pilot/pkg/proxy/envoy/v2/ads_test.go]: @hzxuzhonghu "flake" in what way?  The test emulates the envoy behaviour when it sends a protocol error.

Is this failing with a specific PR?  I'll take a look through the reports later today and see what is going on.

**Pull request merged**